### PR TITLE
Fix 1006 hp8116a

### PR DESCRIPTION
**Pull request by BenediktBurger**
Fixes the hp8116a not using staticmethods (#1006) and improves docstrings in that pr.

### CODE REVIEW

**Comment by BenediktBurger** [comment]
Only from [python 3.10](https://docs.python.org/3/whatsnew/3.10.html) onward, static methods can be called directly, therefore these methods have to go outside of the instrument body.

File: pymeasure/instruments/hp/hp8116a.py [hunk 1]
```diff
@@ -35,6 +35,30 @@
...
+
+def _boolean_control(identifier, state_index, docs, inverted=False, **kwargs):
+    return Instrument.control(
+        'CST', identifier + '%d', docs,
+        validator=strict_discrete_set,
+        values=[True, False],
+        get_process=lambda x: inverted ^ bool(int(x[state_index][1])),
+        set_process=lambda x: int(inverted ^ x),
+        **kwargs
+    )
```
Comment (CasperSchippers) [inline pymeasure/instruments/hp/hp8116a.py:59]: Annoying behaviour in python<3.10 with these static-methods. Is this something we'd like to revert when drop support for pyton<=3.9 (so in about 1.5 year I think)? If so, we might want to mark this as something to change back by then.

File: pymeasure/instruments/hp/hp8116a.py
Comment (CasperSchippers) [inline pymeasure/instruments/hp/hp8116a.py:1]: Nice that you conformize the docstrings. Just noticed a slight inconsistency: for some properties you changed the 'allowed range is X to Y' into '(strict float from X to Y)', while for others you left the range description as it is. Just wondered if this was intentionall.

I'll leave it up to you whether you want to keep it like this or make it consistent.

**Review by CasperSchippers — APPROVED** [review]
Looks good to me. Just some minor remarks.

File: tests/instruments/hp/test_hp8116a.py [hunk 2]
```diff
@@ -31,6 +31,9 @@
 HP8116A.status = property(fget=lambda self: Status(5))
 
 
+init_comm = [(b"CST", b"x" * 87 + b' ,\r\n')]  # communication during init
```
[the remark this answers is not in the document]
Comment (BenediktBurger) [reply tests/instruments/hp/test_hp8116a.py:34]: Done.

File: pymeasure/instruments/hp/hp8116a.py (hunk 1 above)
Comment (BenediktBurger) [reply pymeasure/instruments/hp/hp8116a.py:59]: I added a comment.

File: pymeasure/instruments/hp/hp8116a.py
Comment (BenediktBurger) [reply pymeasure/instruments/hp/hp8116a.py:1]: I made them consistent

**Comment by BenediktBurger** [comment]
@CasperSchippers thanks for your reviews.
I addressed your issues. Feel free to merge, if they suit you.

**Pull request merged**